Model/Sync/Customers/Processor.php

```diff
@@ -440,4 +428,20 @@ public function retryCustomersSync()
             echo 'No customer data to process.' . PHP_EOL;
         }
     }
+
+    /**
+     * @param array $customerSyncData
+     * @param string $currentTime
+     * @return void
+     */
+    private function updateCustomerSyncData(array $customerSyncData, $currentTime)
```

Comment:
I think you should separate into 2 methods:
* One that prepares the data object.
* The second that upserts the data to the DB table (which in this case, is `insertOrUpdateCustomerSyncData`).

Reply:
There's already a function that prepare the data object, it is called few lines before it on each call:
https://github.com/YotpoLtd/magento2-module-messaging/blob/COR-1990-refactor-customers-sync-flows/Model/Sync/Customers/Processor.php#L284
https://github.com/YotpoLtd/magento2-module-messaging/blob/COR-1990-refactor-customers-sync-flows/Model/Sync/Customers/Processor.php#L203
I don't think we should separate it to methods.

Reply:
There's already a function that prepare the data object, it is called few lines before it on each call:
https://github.com/YotpoLtd/magento2-module-messaging/blob/COR-1990-refactor-customers-sync-flows/Model/Sync/Customers/Processor.php#L284
https://github.com/YotpoLtd/magento2-module-messaging/blob/COR-1990-refactor-customers-sync-flows/Model/Sync/Customers/Processor.php#L203
I will return the `$customerSyncData` and call `insertOrUpdateCustomerSyncData`.